nbt/src/main/java/net/kyori/adventure/nbt/TagStringReader.java

```diff
@@ -332,15 +336,18 @@ private int extractRadix(final StringBuilder builder, final String original) {
     if (first == '+' || first == '-') {
       radixPrefixOffset = 1;
     }
-    if (original.startsWith("0b", radixPrefixOffset) || original.startsWith("0B", radixPrefixOffset)) {
+
+    final int radixEndIndex = 2 + radixPrefixOffset;
+
+    if (original.length() > radixEndIndex && (original.startsWith("0b", radixPrefixOffset) || original.startsWith("0B", radixPrefixOffset))) {
```

Comment:
I'd add a comment here like from my PR `// There should be more after '0b'/'0x', else it would be a regular byte tag or string`